ci/cron/src/Main.hs

```diff
@@ -34,42 +34,50 @@ import qualified Network.URI
 import qualified Options.Applicative as Opt
 import Safe (headMay)
 import qualified System.Directory as Directory
+import qualified System.Environment
 import qualified System.Exit as Exit
 import qualified System.IO.Extra as IO
 import qualified System.Process as System
 import qualified Text.Regex.TDFA as Regex
 
-shell_exit_code :: String -> IO (Exit.ExitCode, String, String)
-shell_exit_code cmd = do
-    System.readCreateProcessWithExitCode (System.shell cmd) ""
-
 die :: String -> Int -> String -> String -> IO a
 die cmd exit out err =
     fail $ unlines ["Subprocess:",
-                         cmd,
-                        "failed with exit code " <> show exit <> "; output:",
-                        "---",
-                        out,
-                        "---",
-                        "err:",
-                        "---",
-                        err,
-                        "---"]
+                    cmd,
+                    "failed with exit code " <> show exit <> "; output:",
+                    "---",
+                    out,
+                    "---",
+                    "err:",
+                    "---",
+                    err,
+                    "---"]
 
 shell :: String -> IO String
 shell cmd = System.readCreateProcess (System.shell cmd) ""
 
+proc :: [String] -> IO String
+proc args = System.readCreateProcess (System.proc (head args) (tail args)) ""
+
 shell_ :: String -> IO ()
 shell_ cmd = do
     Control.void $ shell cmd
 
+proc_ :: [String] -> IO ()
+proc_ args = Control.void $ proc args
+
+shell_env_ :: [(String, String)] -> String -> IO ()
+shell_env_ env cmd = do
+    parent_env <- System.Environment.getEnvironment
+    Control.void $ System.readCreateProcess ((System.shell cmd) {System.env = Just (parent_env ++ env)}) ""
```

Comment:
This doesn’t need to be `shell` either right? You can also set `System.env` on the result of `proc` (same return type).

Reply:
I've taken the approach of using `shell` when the entire command is a single string, with no Haskell value interspersed, as I think that reads better. The only use-case we have for setting an env var at the moment fits that, hence `shell_env_` rather than `proc_env_`.

I'm open to changing my mind on that one, but at first glance I prefer
```haskell
shell_env_ [("DAML_SDK_RELEASE_VERSION", show version)] "bazel build //docs:docs"
```
to
```haskell
proc_env_ [("DAML_SDK_RELEASE_VERSION", show version)] ["bazel", "build", "//docs:docs"]
```

Reply:
Happy to keep it. I usually use the guideline to only use `shell` if I actually need a `shell` and not just as a syntactic nicety but no reason to stick to that here.